eva-server/src/main/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2.java

```diff
@@ -127,25 +130,29 @@ private void checkParameters(String variantCoreString, String annotationVepVersi
         }
     }
 
-    private List<VariantWithSamplesAndAnnotation> getVariantEntitiesByParams(String variantCoreString,
-                                                                             String annotationVepVersion,
-                                                                             String annotationVepCacheVersion) throws
-            AnnotationMetadataNotFoundException {
+    private Optional<VariantWithSamplesAndAnnotation> getVariantByCoordinatesAndAnnotationVersion(
+            String variantCoreString,
+                                                                                 String annotationVepVersion,
+                                                                                 String annotationVepCacheVersion)
+            throws AnnotationMetadataNotFoundException, IllegalArgumentException {
         String[] regionId = variantCoreString.split(":");
-        String alternate = (regionId.length > 3) ? regionId[3] : null;
-
+        Optional<String> alternate = Optional.ofNullable(regionId[3]);
+        if (!alternate.isPresent()) {
+            return Optional.ofNullable(null);
```

Comment:
no, I meant that the method *requires* the 4 fields. if the split by ":" doesn't give 4 fields, then throw an exception asking for them.

Try to imagine as a user what would happen with the code you made. you would ask for chr:start:ref, and what would you get? a 404, not found. is that what really happened? does that response explain how the user can make their request work? :)

Reply:
Oh, So should I take in `chromosome, start, reference and alternate` as individual request parameters?

Reply:
I have added the check in the checkParameters functiom :)